diff --git a/node_modules/trezor-connect/lib/core/methods/AbstractMethod.js b/node_modules/trezor-connect/lib/core/methods/AbstractMethod.js
index 6d1e7a4..1f483cf 100644
--- a/node_modules/trezor-connect/lib/core/methods/AbstractMethod.js
+++ b/node_modules/trezor-connect/lib/core/methods/AbstractMethod.js
@@ -248,6 +248,8 @@ var AbstractMethod = /*#__PURE__*/function () {
   _proto.checkFirmwareRange = /*#__PURE__*/function () {
     var _checkFirmwareRange = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3(isUsingPopup) {
       var device, version, model, range, uiPromise, uiResp;
+      var onekeyDeviceVersion;
+
       return _regenerator["default"].wrap(function _callee3$(_context3) {
         while (1) {
           switch (_context3.prev = _context3.next) {
@@ -271,6 +273,7 @@ var AbstractMethod = /*#__PURE__*/function () {
 
             case 5:
               version = device.getVersion();
+              onekeyDeviceVersion = device.features.onekey_version ? device.features.onekey_version.split('.') : version;
               model = version[0];
               range = this.firmwareRange[model];
 
@@ -290,7 +293,7 @@ var AbstractMethod = /*#__PURE__*/function () {
               return _context3.abrupt("return", _constants.UI.FIRMWARE_NOT_SUPPORTED);
 
             case 12:
-              if (!(device.firmwareStatus === 'required' || (0, _versionUtils.versionCompare)(version, range.min) < 0)) {
+              if (!(device.firmwareStatus === 'required' || (0, _versionUtils.versionCompare)(onekeyDeviceVersion, range.min) < 0)) {
                 _context3.next = 14;
                 break;
               }
@@ -298,7 +301,7 @@ var AbstractMethod = /*#__PURE__*/function () {
               return _context3.abrupt("return", _constants.UI.FIRMWARE_OLD);
 
             case 14:
-              if (!(range.max !== '0' && (0, _versionUtils.versionCompare)(version, range.max) > 0)) {
+              if (!(range.max !== '0' && (0, _versionUtils.versionCompare)(onekeyDeviceVersion, range.max) > 0)) {
                 _context3.next = 28;
                 break;
               }
